### PR TITLE
Billpay API clean up 

### PR DESCRIPTION
**Pull request by kiuran**
- Update Service Interface Base Version to v3.14.0
- Update to support Circle 2.0
- Make paths constants 
- Update Hugo build

### CODE REVIEW

File: src/main/java/io/electrum/billpay/utils/Paths.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   public static final String VERSION_NUMBER = "/v4";
+   public static final String BASE_PATH = "/billpay" + VERSION_NUMBER;
+
+   // Account lookup paths
+   public static final String REQUEST_ACCOUNT_INFO = "/accountLookups/{" + PathParams.REQUEST_ID + "}";
+   public static final String REQUEST_TRAFFIC_FINE_INFO = "/accountLookups/traffic/{" + PathParams.REQUEST_ID + "}";
+   public static final String REQUEST_POLICY_INFO = "/accountLookups//policy/{" + PathParams.REQUEST_ID + "}";
+
+   // Payment paths
+   public static final String CREATE_POLICY_PAYMENT = "/policy/{" + PathParams.PAYMENT_ID + "}";
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/billpay/utils/Paths.java]: The /payments prefix is missing for all the payment operations:
CREATE_POLICY_PAYMENT
CREATE_TRAFFIC_FINE_PAYMENT
CREATE_ACCOUNT_PAYMENT 
CONFIRM_PAYMENT 
REVERSE_PAYMENT

File: pom.xml [hunk 2]
```diff
@@ -42,6 +46,9 @@
     <el-version>2.2.4</el-version>
     <swagger-maven-plugin-version>3.1.3</swagger-maven-plugin-version>
     <logback-version>1.0.13</logback-version>
+    <major-version>4</major-version>
```
Comment (VeeneshJeena) [inline pom.xml]: I see that you changed the path for all billpay operations by introducing the prefix "/billpay/v4/".
Surely this would mean it's a major version change to v5.0.0 since exiting people who speak to us must all change their paths.

File: src/main/java/io/electrum/billpay/utils/Paths.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   public static final String CREATE_TRAFFIC_FINE_PAYMENT = "/traffic/{" + PathParams.PAYMENT_ID + "}";
+   public static final String CREATE_ACCOUNT_PAYMENT = "/{" + PathParams.PAYMENT_ID + "}";
+   public static final String CONFIRM_PAYMENT =
+         "/{" + PathParams.PAYMENT_ID + "}/confirmations/{" + PathParams.ADVICE_ID + "}";
+   public static final String REVERSE_PAYMENT =
+         "/{" + PathParams.PAYMENT_ID + "}/reversals/{" + PathParams.ADVICE_ID + "}";
+
+   // Refund paths
+   public static final String CREATE_REFUND = "/refunds/{" + PathParams.REFUND_ID + "}";
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/billpay/utils/Paths.java]: Do you think we should String concat the /refunds/ prefix with the BASE_PATH in the annotation so that each operation doesn't need to prepend it? (similarly for /accountLookups/ and /payments/)
Or maybe define REFUNDS_BASE_PATH etc.. ?
Just a thought.

File: pom.xml (hunk 2 above)
Comment (kiuran) [reply pom.xml]: Hmm will need to check this out because the test server has those paths.

**Comment by kiuran** [comment]
Changes done. I decided to maintain backwards compatibility at this stage.

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
 This page describes changes to the Billpay Service Interface implemented across different releases of the interface.
 
+## v4.6.0
+
+Released TBC
+
+- Update Service Interface Base Version to v3.14.0
```
Comment (VeeneshJeena) [inline src/docs/devguide/hugo/site/content/release-notes.md]: Version should be lowercase "v".

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
+- Update to support Circle 2.0
+- Make paths constants
```
Comment (VeeneshJeena) [inline src/docs/devguide/hugo/site/content/release-notes.md]: Reword this to make the change clearer.
Something like:
Created constants for the paths for each operation
or
Constants for each path/operation are now included
Definitely keep the tense the same as the other entries though.

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 6]
```diff
@@ -1,5 +1,14 @@
 This page describes changes to the Billpay Service Interface implemented across different releases of the interface.
 
+## v4.6.0
+
+Released TBC
+
+- Update Service Interface Base Version to v3.14.0
+- Update to support Circle 2.0
+- Make paths constants 
+- Update Hugo build
```
Comment (VeeneshJeena) [inline src/docs/devguide/hugo/site/content/release-notes.md]: Why? Add a one liner to say why it was updated.
I was going to say lets not add this entry at all, but since all the code in this repo is public, we should say why we changed it. Lets give the readers a one liner on why it was changed, saying what improvements this change brings to their lives (if any).
Yes they can read the commit history, but that isn't always helpful since the descriptions there aren't always helpful (especially if there are multiple commits). Also, the release notes should be a nice summary of whats new and whats improved in the new release.

File: src/main/java/io/electrum/billpay/api/RefundsResource.java [hunk 7]
```diff
@@ -28,27 +31,28 @@
 import io.swagger.annotations.Authorization;
 import io.swagger.annotations.ResponseHeader;
 
-@Path("/refunds/{refundId}")
-@Consumes({ "application/json" })
-@Produces({ "application/json" })
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
 @Api(description = "the payments API", authorizations = { @Authorization("httpBasic") })
```
Comment (VeeneshJeena) [inline src/main/java/io/electrum/billpay/api/RefundsResource.java]: Please fix this description. 
This is the refunds API.

Comment (VeeneshJeena) [reply src/main/java/io/electrum/billpay/api/RefundsResource.java]: Not that these descriptions are really helpful, but lets at least be consistent.

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 8]
```diff
@@ -1,5 +1,14 @@
 This page describes changes to the Billpay Service Interface implemented across different releases of the interface.
 
+## v4.6.0
+
+Released TBC
```
Comment (VeeneshJeena) [inline src/docs/devguide/hugo/site/content/release-notes.md]: Remember to update this TBC.

File: pom.xml [hunk 9]
```diff
@@ -42,6 +46,9 @@
...
+    <minor-version>6</minor-version>
+    <patch-version>0-RC8</patch-version>
```
Comment (RichardFourie) [inline pom.xml]: Remember to make final.

File: src/docs/devguide/hugo/site/content/release-notes.md (hunk 5 above)
Comment (RichardFourie) [reply src/docs/devguide/hugo/site/content/release-notes.md]: Is this still going to happen?

File: src/main/java/io/electrum/billpay/utils/Paths.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
+package io.electrum.billpay.utils;
+
+public class Paths {
```
Comment (RichardFourie) [inline src/main/java/io/electrum/billpay/utils/Paths.java]: Your mission if you choose to accept it is to modify the path constants to a similar way that the aritime and prepaid utility interfaces where done.
